tavern/internal/redirectors/grpc/grpc.go

```diff
@@ -49,10 +50,24 @@ func (r *Redirector) handler(upstream *grpc.ClientConn) grpc.StreamHandler {
 		}
 
 		ctx := ss.Context()
+		// Get the client's remote IP address
+		clientIP := c2.GetClientIP(ctx)
+
 		md, ok := metadata.FromIncomingContext(ctx)
 		if ok {
-			ctx = metadata.NewOutgoingContext(ctx, md)
+			ctx = metadata.NewOutgoingContext(ctx, md.Copy())
 		}
+
+		// Set x-redirected-for header with the client IP
+		if clientIP != "" {
```

Comment:
nit: We could move this into a shared function so that all redirectors don't need to copy paste this